pyenv-win/libexec/libs/shim.py

```diff
@@ -34,10 +34,10 @@
 
         for dir in [bin_dir, scripts_dir]:
             exe_path = os.path.join(dir, exe)
-            if os.path.exists(exe_path):
-                os.environ["PATH"] = os.pathsep.join([bin_dir, scripts_dir, os.environ["PATH"]])
-                result = subprocess.run([exe_path] + sys.argv[1:])
-                sys.exit(result.returncode)
+            os.environ["PATH"] = os.pathsep.join([bin_dir, scripts_dir, os.environ["PATH"]])
+
+    result = subprocess.run([exe_path] + sys.argv[1:])
+    sys.exit(result.returncode)
```

Comment:
This a strange change. You are trying to "not create a subprocess for each active version", but it was not created. I tried to find a suitable exe, then ran it and exited, the loop didn't continue for other versions. Also, you have removed the exe existence check, and `sys.stderr.write(f"{exe} is not found")` below become a dead code. And the final note: python3.10.exe and python3.12.exe aren't exist (only python.exe is), so having both dirs in the path is just useless IMHO.